### DISCUSSION TITLE
Hi after Execution command, download is not begin.

### DISCUSSION THREAD

**Discussion by pllkopcc** [Dineshkarthik/telegram_media_downloader#194] [Q&A]
Hi, thanks for sharing this project, my system is Win10, I manage to follow your instructions till the Execution section, but after running "python3 media_downloader.py" nothing happened, checked  "telegram_media_downloader-master" folder, no new files has being downloaded, do you know what I might do incorrectly?

C:\Users\chen>cd telegram_media_downloader

C:\Users\chen\telegram_media_downloader>python3 media_downloader.py

C:\Users\chen\telegram_media_downloader>python3 media_downloader.py

C:\Users\chen\telegram_media_downloader>

**Comment by rmdes** [comment]
I'm also confused by this part, 
is this log what is supposed to happen ? 
![image](https://user-images.githubusercontent.com/12155003/168283433-f9a0a1ea-d17e-4c61-99c9-085041690712.png)

Am I understanding this correctly that the first run will mark the last read message and then only from that point in time, it will download new media ?

If that's the case, maybe it's worth adding cron job information so that people know that without running the command X times per day, they won't get the file at all ?

**Reply by Dineshkarthik** [reply]
yes, it stores the last message parsed and will read from the next message. the script is stateful and yeah you can add it to the cron job it will work perfectly

**Reply by rmdes** [reply]
Is there a way to know when the download job is done ? 
I'm getting "database locked" so I guess this means download is ongoing ?

**Comment by rmdes** [comment]
Each time I run the command I only update the last ID read, but no matter what is in the channel I don't see anything downloading...
I must be missing something
![image](https://user-images.githubusercontent.com/12155003/168284802-82cdfa60-1af2-4d33-96c5-9cf700e51ea0.png)

**Reply by Dineshkarthik** [reply]
May be you only configured for a particular media type and that media type is not in the new messages.

**Reply by rmdes** [reply]
I think it's because I'm running your code twice from two different directories under the same user. 
I guess If I want to monitor 2 channels I should have each job running under different user ?

**Reply by rmdes** [reply]
It's downloading 
![image](https://user-images.githubusercontent.com/12155003/168294460-59035633-e623-41a1-87c0-66f42fc4c95d.png)

But I'm still confused on how this started on it's own (not configured cron yet) and is actually, by the name of the videos files downloading stuff from the past, back in April

**Reply by Dineshkarthik** [reply]
To monitor 2 channels clone the repo in two different location and configure the config.yaml accordingly (one for each channel) then you can run it in corn and it will download separately.

The script downloads Media file from the very first day the channel is created or the day from which your user is allowed to view messages.

**Reply by rmdes** [reply]
I'm admin of the two channels I want to export, so what you describe is exactly what I did, but I'm stuck, it definitely works on 1 of the channels (image above) but when i switch directory and execute the python command on the second directory/channel I can see it fetching post ID's but it always end up by "doing nothing", no download whatsoever is taking, in fact the video/photos folders are not even created on this second folder. 

i'm wondering if my Python environment could be the cause ?
I'm using 3.10 with Pyenv

**Comment by Dineshkarthik** [comment]
> Hi, thanks for sharing this project, my system is Win10, I manage to follow your instructions till the Execution section, but after running "python3 media_downloader.py" nothing happened, checked "telegram_media_downloader-master" folder, no new files has being downloaded, do you know what I might do incorrectly?
> 
> C:\Users\chen>cd telegram_media_downloader
> 
> C:\Users\chen\telegram_media_downloader>python3 media_downloader.py
> 
> C:\Users\chen\telegram_media_downloader>python3 media_downloader.py
> 
> C:\Users\chen\telegram_media_downloader>

Did you configure the config.py properly with the media type you want to download?
Check if there are actually the media type configured by you is present in the channel ?
